{"uid":"d6f2d7714a8e2cde","name":"test_relation_with_no_database_name","fullName":"tests.integration.new_relations.test_new_relations_1#test_relation_with_no_database_name","historyId":"2ece97842be4c7e2485a69c84df16ec4","time":{"start":1746583955969,"stop":1746583998545,"duration":42576},"description":"Test that a relation with no database name doesn't block the charm.","descriptionHtml":"<p>Test that a relation with no database name doesn't block the charm.</p>\n","status":"broken","statusMessage":"juju.errors.JujuUnitError: Unit in error: database/3","statusTrace":"Traceback (most recent call last):\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: TResult | None = func()\n                             ^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 242, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/threadexception.py\", line 92, in pytest_runtest_call\n    yield from thread_exception_runtest_hook()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/threadexception.py\", line 68, in thread_exception_runtest_hook\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/unraisableexception.py\", line 95, in pytest_runtest_call\n    yield from unraisable_exception_runtest_hook()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/unraisableexception.py\", line 70, in unraisable_exception_runtest_hook\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/logging.py\", line 846, in pytest_runtest_call\n    yield from self._runtest_for(item, \"call\")\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/logging.py\", line 829, in _runtest_for\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/capture.py\", line 898, in pytest_runtest_call\n    return (yield)\n            ^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/skipping.py\", line 257, in pytest_runtest_call\n    return (yield)\n            ^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 174, in pytest_runtest_call\n    item.runtest()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/python.py\", line 1627, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/python.py\", line 159, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.12/asyncio/base_events.py\", line 687, in run_until_complete\n    return future.result()\n           ^^^^^^^^^^^^^^^\n  File \"/root/spread_project/tests/integration/new_relations/test_new_relations_1.py\", line 485, in test_relation_with_no_database_name\n    await ops_test.model.wait_for_idle(apps=APP_NAMES, status=\"active\", raise_on_blocked=True)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/juju/model/__init__.py\", line 3245, in wait_for_idle\n    _raise_for_status(errors, \"error\")\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/juju/model/__init__.py\", line 3136, in _raise_for_status\n    raise error_type(\njuju.errors.JujuUnitError: Unit in error: database/3","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":1,"retriesStatusChange":true,"beforeStages":[{"name":"event_loop","time":{"start":1746581847797,"stop":1746581847798,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1746581847798,"stop":1746581847917,"duration":119},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"tmp_path_factory","time":{"start":1746581847798,"stop":1746581847798,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1746583955969,"stop":1746583955969,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Test that a relation with no database name doesn't block the charm.","status":"broken","statusMessage":"juju.errors.JujuUnitError: Unit in error: database/3","statusTrace":"Traceback (most recent call last):\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: TResult | None = func()\n                             ^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 242, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/threadexception.py\", line 92, in pytest_runtest_call\n    yield from thread_exception_runtest_hook()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/threadexception.py\", line 68, in thread_exception_runtest_hook\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/unraisableexception.py\", line 95, in pytest_runtest_call\n    yield from unraisable_exception_runtest_hook()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/unraisableexception.py\", line 70, in unraisable_exception_runtest_hook\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/logging.py\", line 846, in pytest_runtest_call\n    yield from self._runtest_for(item, \"call\")\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/logging.py\", line 829, in _runtest_for\n    yield\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/capture.py\", line 898, in pytest_runtest_call\n    return (yield)\n            ^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/skipping.py\", line 257, in pytest_runtest_call\n    return (yield)\n            ^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py\", line 174, in pytest_runtest_call\n    item.runtest()\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/python.py\", line 1627, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/_pytest/python.py\", line 159, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.12/asyncio/base_events.py\", line 687, in run_until_complete\n    return future.result()\n           ^^^^^^^^^^^^^^^\n  File \"/root/spread_project/tests/integration/new_relations/test_new_relations_1.py\", line 485, in test_relation_with_no_database_name\n    await ops_test.model.wait_for_idle(apps=APP_NAMES, status=\"active\", raise_on_blocked=True)\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/juju/model/__init__.py\", line 3245, in wait_for_idle\n    _raise_for_status(errors, \"error\")\n  File \"/root/spread_project/.tox/integration/lib/python3.12/site-packages/juju/model/__init__.py\", line 3136, in _raise_for_status\n    raise error_type(\njuju.errors.JujuUnitError: Unit in error: database/3","steps":[],"attachments":[{"uid":"aeddbbe721c268db","name":"log","source":"aeddbbe721c268db.txt","type":"text/plain","size":812}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"abort_on_fail::0","time":{"start":1746583998613,"stop":1746583998618,"duration":5},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test::finalizer","time":{"start":1746584372171,"stop":1746584375640,"duration":3469},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1746584375641,"stop":1746584375641,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1746584375641,"stop":1746584375641,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1746584375641,"stop":1746584375641,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration.new_relations"},{"name":"suite","value":"test_new_relations_1"},{"name":"host","value":"fv-az2027-365"},{"name":"thread","value":"14391-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.new_relations.test_new_relations_1"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"d86760585e5eaff2","status":"unknown","time":{}}],"categories":[{"name":"Test defects","matchedStatuses":[]}],"history":{"statistic":{"failed":0,"broken":1,"skipped":14,"passed":64,"unknown":2,"total":81},"items":[{"uid":"b188b6d52e86f41d","reportUrl":"../4475//#testresult/b188b6d52e86f41d","status":"passed","time":{"start":1746498067583,"stop":1746498120177,"duration":52594}},{"uid":"34ac2aa51bfd8109","reportUrl":"../4462//#testresult/34ac2aa51bfd8109","status":"passed","time":{"start":1746411257232,"stop":1746411310859,"duration":53627}},{"uid":"8323c9ea8ea83cf9","reportUrl":"../4457//#testresult/8323c9ea8ea83cf9","status":"passed","time":{"start":1746324952874,"stop":1746325006129,"duration":53255}},{"uid":"af0506f7df7c3b78","reportUrl":"../4456//#testresult/af0506f7df7c3b78","status":"skipped","statusDetails":"XFAIL reason: aborted\n\n_pytest.outcomes.XFailed: aborted","time":{"start":1746238447895,"stop":1746238447895,"duration":0}},{"uid":"cb08b058d054976b","reportUrl":"../4400//#testresult/cb08b058d054976b","status":"passed","time":{"start":1746152001536,"stop":1746152055055,"duration":53519}},{"uid":"4693b9b6818d56f7","reportUrl":"../4384//#testresult/4693b9b6818d56f7","status":"passed","time":{"start":1746065695695,"stop":1746065749371,"duration":53676}},{"uid":"7bfb9befd98ee35a","reportUrl":"../4377//#testresult/7bfb9befd98ee35a","status":"passed","time":{"start":1745978964925,"stop":1745979018415,"duration":53490}},{"uid":"4e4c6b3113c4c05e","reportUrl":"../4372//#testresult/4e4c6b3113c4c05e","status":"passed","time":{"start":1745893433436,"stop":1745893483931,"duration":50495}},{"uid":"76217bf9d2af16d3","reportUrl":"../4364//#testresult/76217bf9d2af16d3","status":"skipped","statusDetails":"XFAIL reason: aborted\n\n_pytest.outcomes.XFailed: aborted","time":{"start":1745805556227,"stop":1745805556227,"duration":0}},{"uid":"8f6215e193e86816","reportUrl":"../4363//#testresult/8f6215e193e86816","status":"skipped","statusDetails":"XFAIL reason: aborted\n\n_pytest.outcomes.XFailed: aborted","time":{"start":1745720821465,"stop":1745720821465,"duration":0}},{"uid":"cd5916838b7e9f30","reportUrl":"../4360//#testresult/cd5916838b7e9f30","status":"passed","time":{"start":1745633954340,"stop":1745634008559,"duration":54219}},{"uid":"44b1c223d1805bba","reportUrl":"../4353//#testresult/44b1c223d1805bba","status":"passed","time":{"start":1745547627116,"stop":1745547681874,"duration":54758}},{"uid":"bb083a9a1a1decef","reportUrl":"../4345//#testresult/bb083a9a1a1decef","status":"skipped","statusDetails":"XFAIL reason: aborted\n\n_pytest.outcomes.XFailed: aborted","time":{"start":1745461766937,"stop":1745461766937,"duration":0}},{"uid":"44f9cf778dbeb42e","reportUrl":"../4332//#testresult/44f9cf778dbeb42e","status":"passed","time":{"start":1745374945185,"stop":1745374999202,"duration":54017}},{"uid":"ec9564a721c29726","reportUrl":"../4310//#testresult/ec9564a721c29726","status":"passed","time":{"start":1745288335302,"stop":1745288390157,"duration":54855}},{"uid":"c955af7141a49fef","reportUrl":"../4302//#testresult/c955af7141a49fef","status":"passed","time":{"start":1745201755987,"stop":1745201808318,"duration":52331}},{"uid":"e962d3918242006b","reportUrl":"../4299//#testresult/e962d3918242006b","status":"passed","time":{"start":1745115299987,"stop":1745115353167,"duration":53180}},{"uid":"83457280c096f94c","reportUrl":"../4298//#testresult/83457280c096f94c","status":"passed","time":{"start":1745029148325,"stop":1745029200897,"duration":52572}},{"uid":"4080edd22e04db1c","reportUrl":"../4297//#testresult/4080edd22e04db1c","status":"passed","time":{"start":1744942530603,"stop":1744942580411,"duration":49808}},{"uid":"23f9757c14e111c5","reportUrl":"../4293//#testresult/23f9757c14e111c5","status":"passed","time":{"start":1744856101476,"stop":1744856154944,"duration":53468}}]},"tags":["asyncio"]},"source":"d6f2d7714a8e2cde.json","parameterValues":[]}